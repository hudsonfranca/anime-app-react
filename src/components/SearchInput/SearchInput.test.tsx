import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { SearchInput } from '.';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

let wrapper: ShallowWrapper;

beforeEach(() => {
  wrapper = shallow(<SearchInput />);
});

describe('<SearchInput /> redering', () => {
  it('Should render without errors', () => {
    const wrapper = renderer.create(<SearchInput />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render 1 search input container', () => {
    expect(wrapper.find(`[data-test='search-input-container']`)).toHaveLength(
      1
    );
  });

  it('should render 1 input ', () => {
    expect(wrapper.find(`[data-test='input']`)).toHaveLength(1);
  });

  it('should render 1 submit buttont ', () => {
    expect(wrapper.find(`[data-test='submit-button']`)).toHaveLength(1);
  });
});
