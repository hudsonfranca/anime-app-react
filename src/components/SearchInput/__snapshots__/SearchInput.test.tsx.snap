// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> redering Should render without errors 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 button {
  height: 35px;
  width: 40px;
  background-color: var(--white);
  border-radius: 5px;
  border-top-left-radius: unset;
  border-bottom-left-radius: unset;
}

.c0 button:focus {
  outline: none !important;
  border: 1px solid var(--hover1);
  box-shadow: 0 0 7px var(--hover1);
}

.c0 input {
  width: 100%;
  height: 35px;
  border-radius: 5px;
  border: none;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  border-top-right-radius: unset;
  border-bottom-right-radius: unset;
  font-size: 2rem;
  background-color: var(--white);
}

.c0 input:focus {
  outline: none !important;
  border: 1px solid var(--hover1);
  box-shadow: 0 0 7px var(--hover1);
}

.c2 {
  width: 30px;
  height: 30px;
  color: var(--tertiary);
}

<form
  className="c0"
  data-test="search-input-container"
  onSubmit={[Function]}
>
  <input
    data-test="input"
    onChange={[Function]}
    placeholder="Search animes"
    type="text"
    value=""
  />
  <button
    data-test="submit-button"
    type="submit"
  >
    <svg
      aria-hidden="true"
      className="c1 c2"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 18a7.952 7.952 0 004.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0018 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
      />
      <path
        d="M11.412 8.586c.379.38.588.882.588 1.414h2a3.977 3.977 0 00-1.174-2.828c-1.514-1.512-4.139-1.512-5.652 0l1.412 1.416c.76-.758 2.07-.756 2.826-.002z"
      />
    </svg>
  </button>
</form>
`;
